### PR TITLE
Move delete_user and list_users to pipeline architecture

### PR DESCRIPTION
**Pull request by JoshGendein**
Shared UserResponse to replace CreateUserResponse, GetUserResponse, and ReplaceUserResponse
delete_user to pipeline architecture
list_users to pipeline architecture using stream implementation

Continuing #290

### CODE REVIEW

File: sdk/cosmos/src/clients/database_client.rs [hunk 1]
```diff
@@ -91,8 +94,85 @@ impl DatabaseClient {
...
 
     /// List users
-    pub fn list_users(&self) -> requests::ListUsersBuilder<'_, '_> {
-        requests::ListUsersBuilder::new(self)
+    pub fn list_users(
+        &self,
+        ctx: Context,
+        options: ListUsersOptions,
+    ) -> impl Stream<Item = Result<ListUsersResponse, crate::Error>> + '_ {
+        macro_rules! r#try {
```
Comment (rylev) [inline sdk/cosmos/src/clients/database_client.rs:102]: We'll want to move most of this logic to a shared place once this and #340 lands.

File: sdk/cosmos/src/clients/user_client.rs [hunk 2]
```diff
@@ -135,21 +158,6 @@ impl UserClient {
         PermissionClient::new(self, permission_name)
     }
 
-    pub(crate) fn prepare_request_with_user_name(
```
Comment (rylev) [inline sdk/cosmos/src/clients/user_client.rs:138]: I think we actually want to keep this and just make sure that `replace_user` is also using it.

**Review by rylev — APPROVED** [review]
Looks good! We'll want to actually run this code to make sure it still works. In the meantime, I left a few comments on things we might want to improve.

File: sdk/cosmos/src/clients/user_client.rs (hunk 2 above)
Comment (JoshGendein) [reply sdk/cosmos/src/clients/user_client.rs:138]: Good catch, I pushed a change that adds it back and uses it in Get, Replace, and Delete

**Comment by JoshGendein** [comment]
> > tldr: once we resolve the merge conflict, let's merge it?
> 
> We should probably merge #339 first because without it we cannot test if these changes still work against Azure.

@yoshuawuyts @MindFlavor Ok, I fixed the merge conflict so this is ready to be completed. Should this still wait for the auth token PR to be merged?

**Comment by rylev** [comment]
@JoshGendein Sorry for not merging this sooner. Can you rebase again, and then we can merge.
